runtime/doc/news.txt

```diff
@@ -36,6 +36,7 @@ ADDED FEATURES                                                     *news-added*
 
 The following new APIs or features were added.
 
+• Dynamic registration of LSP capabilities. An implication of this change is that checking a client's `server_capabilities` is no longer a sufficient indicator to see if a server supports a feature. Instead use `client.supports_method(<method>)`. It considers both the dynamic capabilities and static `server_capabilities`.
```

Comment:
Thank you very much for working on this, it's really exciting! It would be good to have a good alternative to dynamically react to server capabilities though.

For example, in order to conditionally add a mapping for `textDocument/hover`, I used to check server_capabilities in the `LspAttach` callback, and now I have to override the handler for `client/registerCapability`, and we actually need to do both to account for servers that don't support dynamic registration. Basically, this will lead to some duplication between LspAttach and the handler for `client/registerCapability`, similar to what you had to do for `lsp._set_defaults`.

Personally, I think the duplication isn't too bad (can just create a shared function and call from both places), what's annoying is having to override the handler. It works, but I wonder if you have any plans/ideas on how to improve it?

Thank you again!